scripts/pylib/twister/twisterlib/runner.py

```diff
@@ -87,38 +92,91 @@ def __init__(self, total=0):
 
+    @staticmethod
+    def _find_number_length(n):
+        if n > 0:
+            length = int(log10(n))+1
+        elif n == 0:
+            length = 1
+        else:
+            length = int(log10(-n))+2
+        return length
+
     def summary(self):
-        print("--------------------------------")
-        print(f"Total test suites: {self.total}") # actually test instances
-        print(f"Total test cases: {self.cases}")
-        print(f"Executed test cases: {self.cases - self.skipped_cases}")
-        print(f"Skipped test cases: {self.skipped_cases}")
-        print(f"Completed test suites: {self.done}")
-        print(f"Passing test suites: {self.passed}")
-        print(f"Failing test suites: {self.failed}")
-        print(f"Skipped test suites: {self.skipped_configs}")
-        print(f"Skipped test suites (runtime): {self.skipped_runtime}")
-        print(f"Skipped test suites (filter): {self.skipped_filter}")
-        print(f"Errors: {self.error}")
-        print("--------------------------------")
+        executed_cases = self.cases - self.skipped_cases - self.filtered_cases
+        completed_configs = self.done - self.skipped_filter
+
+        # Find alignment length for aesthetic printing
+        suites_n_length = self._find_number_length(self.total if self.total > self.done else self.done)
+        processed_suites_n_length = self._find_number_length(self.done)
+        completed_suites_n_length = self._find_number_length(completed_configs)
+        skipped_suites_n_length = self._find_number_length(self.skipped_configs)
+        total_cases_n_length = self._find_number_length(self.cases)
+        executed_cases_n_length = self._find_number_length(executed_cases)
+
+        print("--------------------------------------------------")
+        print(f"{'Total test suites: ':<23}{self.total:>{suites_n_length}}") # actually test instances
+        print(f"{'Processed test suites: ':<23}{self.done:>{suites_n_length}}")
+        print(f"├─ {'Filtered test suites (static): ':<37}{self.skipped_filter:>{processed_suites_n_length}}")
+        print(f"└─ {'Completed test suites: ':<37}{completed_configs:>{processed_suites_n_length}}")
+        print(f"   ├─ {'Filtered test suites (at runtime): ':<37}{self.skipped_runtime:>{completed_suites_n_length}}")
+        print(f"   ├─ {'Passed test suites: ':<37}{self.passed:>{completed_suites_n_length}}")
+        print(f"   ├─ {'Failed test suites: ':<37}{self.failed:>{completed_suites_n_length}}")
+        print(f"   └─ {'Errors in test suites: ':<37}{self.error:>{completed_suites_n_length}}")
+        print(f"")
+        print(f"{'Filtered test suites: ':<21}{self.skipped_configs}")
+        print(f"├─ {'Filtered test suites (static): ':<37}{self.skipped_filter:>{skipped_suites_n_length}}")
+        print(f"└─ {'Filtered test suites (at runtime): ':<37}{self.skipped_runtime:>{skipped_suites_n_length}}")
+        print("----------------------      ----------------------")
```

Comment:
```suggestion
        print("--------------------------------------------------")
```

Reply:
Intended - the summary hasn't ended yet, so the divider is different from the opening and closing divider.